### PR TITLE
CASSANDRA-17478

### PR DESCRIPTION
**Pull request by jmckenzie-dev**

### CODE REVIEW

File: src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java [hunk 1]
```diff
@@ -297,7 +298,9 @@ else if (replication != null)
                        .forEach(dc -> options.putIfAbsent(dc, defaultReplicas.toParseableString()));
         }
 
-        options.values().removeAll(Collections.singleton("0"));
+        // We do not filter out "0" due to the AUTH_KEYSPACE validation in validateExpectedOptions.
+        // The previousOptions are unavailable to that check so it is not possible to verify that all valid
+        // datacenters have an explicit configuration (even if it is 0).
```
Comment (jonmeredith) [inline src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java]: I don't think this is needed in general, just some specific operational checks that should be changed.

**Review by jonmeredith — APPROVED** [review]
Approved with recommendation to re-instate cleaning up DCs with RF 0

File: src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java (hunk 1 above)
Comment (jmckenzie-dev) [reply src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java]: You don't think the comment is needed, the filtering out of "0", the removal of filtering out of "0" here...?

Or something else entirely? ;)

Comment (jonmeredith) [reply src/java/org/apache/cassandra/locator/NetworkTopologyStrategy.java]: I think the old code to filter out "0" should be retained, and the comment is no longer needed if you keep it.

**Pull request closed without merging**